### ISSUE TITLE
Unable to sync with --no-warp

### ISSUE THREAD

**Issue by abitmore** [openethereum/openethereum#300]
- **OpenEthereum version (>=3.1.0)**: 3.1.1 / 3.2.0-rc.1
- **Operating system**: Linux (Ubuntu 16.04 / 18.04 / 20.04)
- **Installation**: zip download from release page
- **Fully synchronized**: no
- **Network**: ethereum
- **Restarted**: yes

>_Your issue description goes here below. Try to include **actual** vs. **expected behavior** and **steps to reproduce** the issue._

I've been with the issue for months. Tried on Ubuntu 16.04, 18.04 and 20.04, several versions (all 3.x) of binaries downloaded from Github (haven't tried building by myself). Was always trying on AWS, with 30303 port open (configured in security options) or not.

I start openethereum like this (since parity works with the same environment).
```
./openethereum --no-warp --db-path mypath
```

It always stops syncing a short time after started. The block number when it stops at seems random. Typical console output:
```
2021-03-07 16:18:52 UTC Starting OpenEthereum/v3.1.1-stable-15b5581-20210119/x86_64-linux-gnu/rustc1.49.0
2021-03-07 16:18:52 UTC Keys path ...
2021-03-07 16:18:52 UTC DB path ...
2021-03-07 16:18:52 UTC State DB configuration: fast
2021-03-07 16:18:52 UTC Operating mode: active
2021-03-07 16:18:52 UTC Configured for Ethereum using Ethash engine
2021-03-07 16:18:52 UTC Listening for new connections on 127.0.0.1:8546.
2021-03-07 16:18:53 UTC Updated conversion rate to Ξ1 = US$1667.47 (2855766 wei/gas)
2021-03-07 16:18:57 UTC Public node URL: enode://xxxxx...xxxxx@x.x.x.x:30303
2021-03-07 16:18:57 UTC Syncing    #2085 0xcf53…c73c   411.00 blk/s    0.0 tx/s    0.0 Mgas/s   8711+    0 Qed LI:#10965    5/25 peers   2 MiB chain 14 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:02 UTC Syncing    #3856 0xa367…9c54   354.20 blk/s    0.0 tx/s    0.0 Mgas/s  24878+ 2125 Qed LI:#30861    7/25 peers   4 MiB chain 40 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:07 UTC Syncing    #9973 0x15f1…a789  1223.40 blk/s    0.0 tx/s    0.0 Mgas/s  26093+    0 Qed LI:#36068    7/25 peers   6 MiB chain 39 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:12 UTC Syncing   #14248 0x8114…f9f6   854.83 blk/s    0.0 tx/s    0.0 Mgas/s  26771+    0 Qed LI:#41021    8/25 peers   6 MiB chain 40 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:17 UTC Syncing   #15683 0xd062…3768   292.62 blk/s    0.0 tx/s    0.0 Mgas/s  25253+ 3638 Qed LI:#44578    8/25 peers   3 MiB chain 43 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:22 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s  19677+ 9216 Qed LI:#44578    9/25 peers   3 MiB chain 43 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:27 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s  14577+14316 Qed LI:#44578    9/25 peers   3 MiB chain 42 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:32 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s   9236+19657 Qed LI:#44578    9/25 peers   3 MiB chain 42 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:37 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s   3485+25405 Qed LI:#44578    9/25 peers   3 MiB chain 42 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:42 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+28895 Qed LI:#44578    9/25 peers   3 MiB chain 41 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:47 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+28895 Qed LI:#44578   10/25 peers   3 MiB chain 41 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:52 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+28895 Qed LI:#44578   12/25 peers   3 MiB chain 41 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:19:57 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+28895 Qed LI:#44578   12/25 peers   3 MiB chain 41 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
2021-03-07 16:20:02 UTC Syncing   #15683 0xd062…3768     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+28895 Qed LI:#44578   13/25 peers   3 MiB chain 41 MiB queue  RPC:  0 conn,    0 req/s,    0 µs
...
```

After restart it continues syncing for a while then stops again.

Any idea?

(Update: just tried 3.2.0-rc.1, same issue)

**Comment by rakita** [comment]
I tried it with v3.2.0-rc.1, it works for me

**Comment by abitmore** [comment]
Just tried the latest v3.2.0 release on a Ubuntu 20.04 VPC with no luck.

Steps to reproduce:
* start a default Ubuntu 20.04 / `i3.large` instance on AWS EC2
* ssh into the VPC, run commands
```
wget https://github.com/openethereum/openethereum/releases/download/v3.2.0/openethereum-linux-v3.2.0.zip
sudo apt install unzip
unzip openethereum-linux-v3.2.0.zip
chmod u+x openethereum
./openethereum --no-warp -d data
```

It stalls after a while.

Here is a log file when started with
```
./openethereum -l debug --no-warp -d /data/oe --log-file=/data/oe.log
```
[oe.log.gz](https://github.com/openethereum/openethereum/files/6101083/oe.log.gz)

```
/dev/nvme0n1    436G  119M  414G   1% /data
```

**Comment by abitmore** [comment]
It seems I need to compile the code? I've never tried compiling. Will try when got time. Thanks.

**Comment by rakita** [comment]
[openethereum.zip](https://github.com/openethereum/openethereum/files/6109609/openethereum.zip)

here is binary from my machine

**Comment by abitmore** [comment]
Thanks. It seems it works. I think that line of code does the trick:
```
self.block_queue.resignal_verification();
```
In the log file I found quite some "block_queue is empty" messages. Do you still want the log file?

BTW I found the code quite easy to compile on Ubuntu. Thanks again.

Please feel free to close the issue when the patch is merged.

**Comment by rakita** [comment]
yea, can you still share logs, just to recheck things?
but as you said `resignal_verification` is probably the solver for this issue.

**Comment by abitmore** [comment]
Partial log is here (strangely it seems Github doesn't allow me upload the full log).
[oe1a.log.gz](https://github.com/openethereum/openethereum/files/6117934/oe1a.log.gz)
Wish it helps.

**Comment by roninkaizen** [comment]
@abitmore 
did you ever tried to reduce mem, with --cache-size=8192 for instance?
i remember two sessions about 9 weeks ago where i encountered the same with the downloaded binaries,
but on second try with limits set, i went "through" since then.
would be really interested in your answer, if you like

**Comment by abitmore** [comment]
I haven't tried to change cache size. Will try and report back. Thanks.

**Comment by rakita** [comment]
No need to set cache-size, it is unrelated.

**Comment by abitmore** [comment]
OK. I haven't got time to try yet. :)

**Comment by voronindev** [comment]
Latest image at docker hub are works.
`image: openethereum/openethereum:nightly`
`command: --no-warp --cache-size 4096 --scale-verifiers --no-secretstore --jsonrpc-interface all --chain ropsten --tracing on`

**Comment by roninkaizen** [comment]
![grafik](https://user-images.githubusercontent.com/17592883/111537875-43e17980-876c-11eb-802a-d7dd1ee9bc98.png)
soo-
started mainnet again, fresh :)

**Comment by rakita** [comment]
fix pushed to `v3.2.1` version. @abitmore thank you for your support!
